File: jce.Front/src/app/@core/data/models/catalog/catalog.ts
import {CatalogProduct} from '../products/catalogProduct';
import {CatalogGood} from '../products/catalogGood';

class Catalog {

  id: number;
  ceId: number;
  indexId: string;
  catalogType: number;
  isActif: boolean;
  expirationDate: Date;
  displayPrice: boolean;
  productChoiceQuantity: number;
  catalogChoiceTypeId: number;
  booksQuantity: number;
  toysQuantity: number;
  subscriptionQuantity: number;

  catalogGoods: CatalogGood[];

  createdBy: string;
  updatedBy: string;
  // constructor(id: number, ceId: number, indexId: string, catalogType: number, isActif: boolean,
  //             expirationDate: string, displayPrice: boolean, productChoiceQty: number, expiration: Date,
  //             isUniqueChoice: boolean, isMultipleChoice: boolean, isCumulatedProductChoice: boolean,
  //             booksQty: number, toysQty: number, subQty: number, goods: CatalogGood[],
  //             createdBy: string, updatedBy: string) {}

}

class FormCatalogModel {
  constructor(public id: number,
              public ceId: number,
              public catalogType: number,
              public isActif: boolean,
              public expirationDate: Date,
              public displayPrice: boolean,
              public productChoiceQuantity: number,
              public catalogChoiceTypeId: number,
              public booksQuantity: number,
              public  toysQuantity: number,
              public  subscriptionQuantity: number,
              public  catalogGoods: CatalogGood[]) {
  }
}

export {Catalog, FormCatalogModel};

File: jce.Server/jce.Common/Entites/JceDbContext/Catalog.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;
using jce.Common.Core;

namespace jce.Common.Entites
{
    [Table("Catalogs")]
    public class Catalog : BaseEntity
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [ForeignKey("CEs")]
        public int CeId { get; set; }
        public string IndexId { get; set; }
        [Required]
        public int CatalogType { get; set; }
        public bool? IsActif { get; set; }
        public DateTime ExpirationDate { get; set; }
        public bool DisplayPrice { get; set; }

        public int CatalogChoiceTypeId { get; set; }
        public int? ProductChoiceQuantity { get; set; }

        public int? BooksQuantity { get; set; }
        public int? ToysQuantity { get; set; }
        public int? SubscriptionQuantity { get; set; }

        public ICollection<CatalogGood> CatalogGoods { get; set; }

        public Catalog()
        {
            CatalogGoods = new Collection<CatalogGood>();
        }
    }
}
